sites/public/pages/finder.tsx

```diff
@@ -0,0 +1,211 @@
+          labels={["Housing Needs", "Accessibility", "Building Types"]}
+          mounted={true}
+          style="bar"
+        ></ProgressNav>
+      </div>
+    )
+  }
+
+  const nextQuestion = () => {
+    const userSelections = getValues()[formData[currentStep]["fieldGroupName"]]
+    formData[currentStep]["fields"].forEach(
+      (field) => (field["selected"] = userSelections.includes(field))
+    )
+    setCompletedSteps(completedSteps + 1)
+    setCurrentStep(currentStep + 1)
+  }
+  const previousQuestion = () => {
+    const userSelections = getValues()[formData[currentStep]["fieldGroupName"]]
+    formData[currentStep]["fields"].forEach(
+      (field) => (field["selected"] = userSelections.includes(field))
+    )
+    setCompletedSteps(completedSteps - 1)
+    setCurrentStep(currentStep - 1)
+  }
+
+  return (
+    <Layout>
+      <Form onSubmit={handleSubmit(onSubmit)} className="bg-gray-300 border-t border-gray-450">
+        <div className="md:mb-20 md:mt-12 mx-auto max-w-5xl">
+          {ProgressHeader(currentStep, completedSteps)}
+          <FormCard>
+            {activeQuestion && (
+              <>
+                <div className="px-20">
+                  <div className="pt-12">
+                    <div className="text-3xl pb-4">
+                      {t(`finder.${activeQuestion["fieldGroupName"]}.question`)}
+                    </div>
+                    <div className="pb-8 border-b border-gray-450">
+                      {t("finder.questionSubtitle")}
+                    </div>
+                  </div>
+                  <div className="py-8">
+                    <p className="pb-4">{t("finder.multiselectHelper")}</p>
+                    <div className="finder-grid">
+                      {activeQuestion["fields"]?.map((field) => {
+                        return (
+                          <div className="finder-grid__field">
+                            <Field
+                              name="bedRoomSize"
+                              register={register}
+                              id={FrontendListingFilterStateKeys[field.label]}
+                              label={t(`listingFilters.bedroomsOptions.${field.translation}`)}
+                              key={FrontendListingFilterStateKeys[field.label]}
+                              type="checkbox"
+                              inputProps={{
+                                value: FrontendListingFilterStateKeys[field.label],
+                                defaultChecked: field.selected,
+                              }}
+                              bordered
+                            />
+                          </div>
+                        )
+                      })}
+                    </div>
+                  </div>
+                </div>
+
+                <div className="bg-gray-300 flex flex-row-reverse justify-between py-8 px-20">
+                  {/* change as more questions added */}
```

Comment:
I had to adjust the handling of state a bit for this one (formData as state rather than just active question), but I agree that it is better with a length check!